### DISCUSSION TITLE
Where are the `date!()` and `time!()` docs?

### DISCUSSION THREAD

**Discussion by detly** [time-rs/time#331] [Question] [answered]
Do the `date!()` and `time!()` macros have documentation? I haven't been able to find it on [docs.rs](https://docs.rs/time/0.2.27/time/macro.date.html). What do they take?

Alternatively, is there a way to construct a `PrimitiveDateTime` or `OffsetDateTime` from compile-time-known values directly?

**Comment by jhpratt** [comment, marked as answer]
Go to the [`macros` module](https://docs.rs/time/0.2.27/time/macros/index.html), which is the preferred way to import the macros. For some reason rustdoc doesn't copy documentation when re-exported.

With regard to `PrimitiveDateTime` and `OffsetDateTime`, it is not possible in v0.2 to construct them directly. You'll have to use per-existing methods on the structs. If you're using the alpha (which it doesn't look like), you can use the `datetime!` macro.
